### ISSUE TITLE
Don't quit process on SMTP errors

### ISSUE THREAD

**Issue by davewasmer** [maildev/maildev#298]
I'm using maildev programmatically (i.e. I boot it up alongside my dev server in the same process), and ran into an issue where a misconfigured port number cause an error while maildev tried to start up.

The result was that my app would start, and then shut down with no error messages or explanation (I have `silent: true` to disable the startup messages since I'm printing out my own startup messages).

I think ideally maildev would simply shut down, but would forcibly close the process. If maildev is the only thing running, then once the SMTP and web servers release their ports, Node will quit the process automatically. But for users like myself, we can choose to keep the process running and handle the failure in a different manner.

**Comment by heldchen** [comment]
I think this is fixed in 2.0.0, see https://github.com/maildev/maildev/issues/287

**Comment by davewasmer** [comment]
I don't think so, unfortunately. I haven't tested that PR, but even if it happens to fix the error around a bad port number, it looks like maildev still calls `process.exit()` when other errors are encountered (see my links above).

So even if a misconfigured port no longer results in the `process.exit()` code running, I assume something else could, and that seems like the root problem to me: maildev shouldn't assume it's the only thing running in the process, and should instead simply close it's own servers/ports but not forcible quit the process.

**Comment by heldchen** [comment]
no, it prevents maildev from exiting after any smtp protocol errors encountered in the upstream smtp npm packages:
```client.on('error', function(e) { logger.error('Caught SMTP protocol error: ', e) })```

haven't had a single process quit since adding this a few months ago.

**Comment by davewasmer** [comment]
I understand that, but it seems that maildev still includes the following lines:

https://github.com/maildev/maildev/blob/master/lib/mailserver.js?rgh-link-date=2020-04-28T20%3A46%3A55Z#L232

https://github.com/maildev/maildev/blob/master/index.js?rgh-link-date=2020-04-28T20%3A46%3A55Z#L80

Both of those lines seem inappropriate - I can't think of any good reason why maildev should ever be emitting fake SIGTERM events or calling process.exit.

**Comment by jeremy-w** [comment]
this behavior - where maildev assumes it owns the whole process - is proving problematic for me, as well. i'm integrating maildev into a project, and i noticed it was taking out the test process well before all tests had run: the usual test scenario in this project is to boot a server per test, which tries to boot a maildev instance per test, which fails out on port binding, because the first instance already snagged it. the failing instance then takes the whole test process with it.

**Comment by djfarrelly** [comment]
This is resolved for programmatic/embedded use in 3.0. The library packages an embedder imports — `@maildev/core`, `@maildev/api` (`createAPIServer`), `@maildev/smtp` (`createSMTPServer`) — contain **zero** `process.exit()` calls. Startup failures (bad port, `EADDRINUSE`) now reject the `start()`/`listen()` promise, and runtime SMTP errors are emitted as an `'error'` event, so your host process stays in control.

`process.exit()` now lives only in the entrypoints you wouldn't import: the CLI (`packages/cli`), the dev harness (`packages/api/src/dev.ts`), and the MCP stdio runner.

When embedding, `await` the server start (to catch bind errors) and attach an `'error'` listener to the SMTP server so protocol errors don't surface as an uncaught exception. Closing as fixed — please reopen if you still see a forced exit from the library surface on the current release.
